### PR TITLE
Remove leading slash from command block commands

### PR DESCRIPTION
**Pull request by mathiascode**
Vanilla allows you to include the slash.

### CODE REVIEW

**Comment by madmaxoft** [comment]
The code says "initial", rather than "trailing".

**Comment by mathiascode** [comment]
Looks like my brain shut down. I'll correct the mistake. :P

File: src/BlockEntities/CommandBlockEntity.cpp [hunk 1]
```diff
@@ -178,6 +178,12 @@ void cCommandBlockEntity::Execute()
 		}
 	} CmdBlockOutCb(this);
 
+	// Remove leading slash if one exists, since console commands don't use them
+	if (m_Command[0] == '/')
```
Comment (madmaxoft) [inline src/BlockEntities/CommandBlockEntity.cpp]: Can this codepath be taken even when `m_Command` is empty? Then this would fail.

**Comment by madmaxoft** [comment]
For some reason I can't put this into line comment. Your check won't work, acessing the zero-th element of an empty string will fail.

Instead: `if (!m_Command.empty() && (m_Command[0] == '/'))`
Or even better, add a check to the top of the entire function, if the command is empty, just bail out completely.

**Comment by mathiascode** [comment]
This isn't a good solution. It doesn't handle double slash commands properly.

**Comment by madmaxoft** [comment]
Does vanilla allow (and discard) multiple slashes in front of commands?

**Comment by Pokechu22** [comment]
No, vanilla will only strip at most one slash from a command, and that will cause issues with worldedit commands in command blocks (I don't remember how that's worked around; it might be by prefixing it with a namespace i.e. `worldedit:/set` or `/worldedit:/set`).

File: src/BlockEntities/CommandBlockEntity.cpp [hunk 2]
```diff
@@ -175,21 +180,30 @@ void cCommandBlockEntity::Execute()
 		{
 			// Overwrite field
 			m_CmdBlock->SetLastOutput(cClientHandle::FormatChatPrefix(m_CmdBlock->GetWorld()->ShouldUseChatPrefixes(), "SUCCESS", cChatColor::Green, cChatColor::White) + a_Text);
+			m_CmdBlock->GetWorld()->BroadcastBlockEntity(m_CmdBlock->GetPos());
 		}
 	} CmdBlockOutCb(this);
 
+	AString RealCommand = m_Command;
+
+	// Remove trailing slash if it exists, since console commands don't use them
```
Comment (bearbin) [inline src/BlockEntities/CommandBlockEntity.cpp]: The first character can never be a _trailing_ slash, it may only be _leading_.

Comment (mathiascode) [reply src/BlockEntities/CommandBlockEntity.cpp]: Oops, looks like I reverted my comment to a previous version. Will fix.

**Pull request merged**